docker/Dockerfile.replication

```diff
@@ -0,0 +1,41 @@
+FROM python:3.11.6-slim-bookworm
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Update and install OS dependencies
+RUN apt-get -y update && \
+    apt-get -y upgrade && \
+    apt-get -y install --no-install-recommends git && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install required python build dependencies
+RUN pip install --upgrade --no-cache-dir pip setuptools wheel uv
+
+# Create the build directory
+WORKDIR /build
+COPY . /build
+
+# Install requirements
+RUN uv pip install --no-cache-dir --system cassandra-driver psycopg2-binary
+RUN uv pip install --no-cache-dir --system -r requirements.txt
+
+# Install the package
+RUN uv pip install --no-cache-dir --system --no-deps .
+
+# Setup the application scripts
+WORKDIR /app
+
+# Install sdm_schemas
+RUN git clone https://github.com/lsst/sdm_schemas.git
```

Comment:
> Maybe add `--depth=0` option to avoid cloning full history?

Good idea - will do.

> 
> Just a general comment for the future - we may need to install a specific version of `sdm_schemas` that matches the version of the database schema. It may be even better to delay or avoid cloning sdm_schemas into the image, to be able to use the same image with different schema version. Something to work out later.

I think sdm_schemas should be made pip installable - then we can use it in the requirements, including a specific tag or branch. I am working on this. For now, I'll just leave this as is.